packages/@ember/-internals/glimmer/lib/component-managers/outlet.ts

```diff
@@ -46,6 +46,7 @@ export interface OutletDefinitionState {
   template: Template;
   controller: unknown;
   model: unknown;
+  component: object | undefined;
```

Comment:
The choice of type here reflects that a component is anything with a component manager, and having a component manager requires being a weakmap key.